src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -0,0 +1,620 @@
+            AddIfNotNull(syncMethods, methods);
+
+            methods = namedType?.GetMembers(LongCount).OfType<IMethodSymbol>().Where(m => m.Parameters.Length <= 2);
+            AddIfNotNull(syncMethods, methods);
+
+            namedType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemLinqQueryable);
+            methods = namedType?.GetMembers(Count).OfType<IMethodSymbol>().Where(m => m.Parameters.Length <= 2);
+            AddIfNotNull(syncMethods, methods);
+
+            methods = namedType?.GetMembers(LongCount).OfType<IMethodSymbol>().Where(m => m.Parameters.Length <= 2);
+            AddIfNotNull(syncMethods, methods);
+
+            namedType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftEntityFrameworkCoreEntityFrameworkQueryableExtensions);
+            methods = namedType?.GetMembers(CountAsync).OfType<IMethodSymbol>().Where(m => m.Parameters.Length <= 2);
+            AddIfNotNull(asyncMethods, methods);
+
+            methods = namedType?.GetMembers(LongCountAsync).OfType<IMethodSymbol>().Where(m => m.Parameters.Length <= 2);
+            AddIfNotNull(asyncMethods, methods);
+
+            namedType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemDataEntityQueryableExtensions);
+            methods = namedType?.GetMembers(CountAsync).OfType<IMethodSymbol>().Where(m => m.Parameters.Length <= 2);
+            AddIfNotNull(asyncMethods, methods);
+
+            methods = namedType?.GetMembers(LongCountAsync).OfType<IMethodSymbol>().Where(m => m.Parameters.Length <= 2);
+            AddIfNotNull(asyncMethods, methods);
+
+            if (syncMethods.Count > 0 || asyncMethods.Count > 0)
+            {
+                context.RegisterOperationAction(operationContext => AnalyzeInvocationOperation(operationContext, syncMethods, asyncMethods), OperationKind.Invocation);
+            }
+
+            context.RegisterOperationAction(AnalyzePropertyReference, OperationKind.PropertyReference);
+        }
+
+        private static void AnalyzeInvocationOperation(OperationAnalysisContext context, HashSet<IMethodSymbol> syncMethods, HashSet<IMethodSymbol> asyncMethods)
+        {
+            var invocationOperation = (IInvocationOperation)context.Operation;
+
+            // For C#, TargetMethod returns the actual extension method that we are looking for.
+            // For VB, TargetMethod shows as a member of the extended class, in order to get the static extension method, we need to call ReducedFrom.
+
+            // We use OriginalDefinition to normalize the method to its generic version. For VB we don't use it since I assume ReducedFrom already goes directly to the generic version.
+            IMethodSymbol originalDefinition = invocationOperation.Language == LanguageNames.VisualBasic ?
+                invocationOperation.TargetMethod.ReducedFrom :
+                invocationOperation.TargetMethod.OriginalDefinition;
```

Comment:
What if we use `IMethodSymbol originalDefinition = (invocationOperation.TargetMethod.ReducedFrom ?? invocationOperation.TargetMethod).OriginalDefinition` to avoid language specific checks?